hadoop-tools/hadoop-fs2img/src/main/java/org/apache/hadoop/hdfs/server/namenode/UGIResolver.java

```diff
@@ -132,4 +138,134 @@ public FsPermission permission(FileStatus s) {
     return s.getPermission();
   }
 
+  private long resolve(AclStatus aclStatus) {
+    return buildPermissionStatus(
+        user(aclStatus), group(aclStatus), permission(aclStatus).toShort());
+  }
+
+  /**
+   * Get the locally mapped user for external {@link AclStatus}.
+   *
+   * @param aclStatus AclStatus on external store.
+   * @return locally mapped user name.
+   */
+  public String user(AclStatus aclStatus) {
```

Comment:
getUser()
Is this static?

Reply:
Why do we have this just for doing a get?

Reply:
The user mapping behavior can be different in a subclass. For e.g. In this case `SingleUGIResolver` maps all users to a single user.